src/Board.py

```diff
@@ -45,10 +49,46 @@ def displayBoard(board):
         line = chr(x + UNICODE_FOR_A_CHAR)
 
         for i in range(len(board)):
-            
-            if board[x][i]:
+            if board[x][i] == SQUARE_SHIP:
+                line += " S "
+            elif board[x][i] == SQUARE_FAILED_SHOT:
+                line += " O "
+            elif board[x][i] == SQUARE_SUCCESS_SHOT:
                 line += " X "
             else:
                 line += "   "
 
         print(line)
+
+def shoot(coord, boards, currentPlayer):
+    boardsCopy = deepcopy(boards)
+    hit = 0
+
+    if (currentPlayer == 1):
+        if (shipAt(coord, boards["player2"]["primary"])):
+            boardsCopy["player1"]["opponent"] = updateBoard(boards["player1"]["opponent"], coord, SQUARE_SUCCESS_SHOT)
+            hit = 1
+        else:
+            boardsCopy["player1"]["opponent"] = updateBoard(boards["player1"]["opponent"], coord, SQUARE_FAILED_SHOT)
+    else:
+        if (shipAt(coord, boards["player1"]["primary"])):
```

Comment:
Looks like there is an opportunity to refactor the code using a function (lines from 68 to 72 and 74 to 74 are similar).